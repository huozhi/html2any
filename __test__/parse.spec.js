const fs = require('fs')
const html1 = require('./mock/html1')
const html2 = require('./mock/html2')
const tokenize = require('../src/tokenize')
const parse = require('../src/parse')

const parser = (content) => parse(tokenize(content))

it('parse correctly', () => {
  const tks = tokenize(html1)
  expect(parse(tks)).toMatchSnapshot()
})

it('parse correctly', () => {
  const tks = tokenize(html2)
  expect(parse(tks)).toMatchSnapshot()
})

it('parse xml correctly', () => {
  const content = fs.readFileSync(__dirname + '/mock/mpd-1.xml', 'utf8')
  const tks = tokenize(content)
  expect(parse(tks)).toMatchSnapshot()
})
