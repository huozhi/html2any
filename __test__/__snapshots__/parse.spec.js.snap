// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse correctly 1`] = `
Array [
  Object {
    "attributes": Object {},
    "children": Array [
      Object {
        "attributes": Object {},
        "children": undefined,
        "name": "hr",
      },
      Object {
        "attributes": Object {
          "data-index": "2",
          "src": "a.jpg",
        },
        "children": undefined,
        "name": "img",
      },
      Object {
        "attributes": Object {
          "id": "x34",
        },
        "children": Array [
          "1234 &amp;",
        ],
        "name": "p",
      },
      Object {
        "attributes": Object {
          "class": "d3 d2",
          "id": "container",
        },
        "children": Array [
          Object {
            "attributes": Object {
              "data-id": "xx",
              "readonly": true,
              "value": "1 2 3",
            },
            "children": undefined,
            "name": "input",
          },
        ],
        "name": "div",
      },
    ],
    "name": "div",
  },
]
`;

exports[`parse correctly 2`] = `
Array [
  Object {
    "attributes": Object {},
    "children": Array [
      Object {
        "attributes": Object {},
        "children": Array [
          Object {
            "attributes": Object {},
            "children": undefined,
            "name": "br",
          },
        ],
        "name": "p",
      },
      Object {
        "attributes": Object {
          "data-mce-style": "text-align: center; font-size: 15px;",
          "style": "text-align: center; font-size: 15px;",
        },
        "children": Array [
          Object {
            "attributes": Object {
              "alt": "Tiny M C E",
              "data-mce-src": "images/glyph-tinymce@2x.png",
              "height": "97",
              "src": "https://www.tinymce.com/images/glyph-tinymce@2x.png",
              "width": "110",
            },
            "children": undefined,
            "name": "img",
          },
        ],
        "name": "p",
      },
      Object {
        "attributes": Object {
          "style": "text-align: center; color: #666; font-size: 15px; font-family: avenir;",
        },
        "children": Array [
          "This demo includes these TinyMCE Premium",
          Object {
            "attributes": Object {
              "class": "mce-spellchecker-word",
              "data-mce-bogus": "1",
              "data-mce-index": "0",
              "data-mce-word": "Plugins",
            },
            "children": Array [
              "Plugins",
            ],
            "name": "span",
          },
          ":",
          Object {
            "attributes": Object {
              "data-mce-href": "powerpaste",
              "data-mce-linkchecker-status": "valid",
              "href": "https://www.tinymce.com/powerpaste",
            },
            "children": Array [
              "PowerPaste",
            ],
            "name": "a",
          },
          "(improved Word paste),",
          Object {
            "attributes": Object {
              "data-mce-href": "accessibility-checker",
              "data-mce-linkchecker-status": "valid",
              "href": "https://www.tinymce.com/accessibility-checker",
            },
            "children": Array [
              "Accessibility Checker",
            ],
            "name": "a",
          },
          ",",
          Object {
            "attributes": Object {
              "data-mce-href": "spell-checker-pro",
              "data-mce-linkchecker-status": "valid",
              "href": "https://www.tinymce.com/spell-checker-pro",
            },
            "children": Array [
              "Spell Checker Pro",
            ],
            "name": "a",
          },
          "&amp; file management with",
          Object {
            "attributes": Object {
              "data-mce-href": "http://www.moxiemanager.com",
              "data-mce-linkchecker-status": "valid",
              "href": "http://www.moxiemanager.com",
            },
            "children": Array [
              "MoxieManager",
            ],
            "name": "a",
          },
          ".",
        ],
        "name": "p",
      },
      Object {
        "attributes": Object {
          "data-mce-style": "text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;",
          "style": "text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;",
        },
        "children": Array [
          "TinyMCE is a platform indepen",
          Object {
            "attributes": Object {},
            "children": Array [
              "dent web-",
            ],
            "name": "strong",
          },
          "based JavaScript HTML WYSIWYG",
          Object {
            "attributes": Object {},
            "children": undefined,
            "name": "br",
          },
          "editor control released as open source",
          Object {
            "attributes": Object {},
            "children": Array [
              "under",
            ],
            "name": "em",
          },
          "LGPL.",
        ],
        "name": "p",
      },
      Object {
        "attributes": Object {
          "data-mce-style": "text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;",
          "style": "text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;",
        },
        "children": Array [
          Object {
            "attributes": Object {},
            "children": Array [
              "TinyMCE enables you to convert HTML",
              Object {
                "attributes": Object {
                  "class": "mce-spellchecker-word",
                  "data-mce-bogus": "1",
                  "data-mce-index": "0",
                  "data-mce-word": "textarea",
                },
                "children": Array [
                  "textarea",
                ],
                "name": "span",
              },
              "fields or other HTML elements to editor instances.",
            ],
            "name": "em",
          },
        ],
        "name": "p",
      },
      Object {
        "attributes": Object {},
        "children": Array [
          Object {
            "attributes": Object {},
            "children": undefined,
            "name": "br",
          },
        ],
        "name": "p",
      },
    ],
    "name": "div",
  },
]
`;

exports[`parse xml correctly 1`] = `
Array [
  Object {
    "attributes": Object {
      "mediaPresentationDuration": "PT0H3M1.63S",
      "minBufferTime": "PT1.5S",
      "profiles": "urn:mpeg:dash:profile:isoff-on-demand:2011",
      "type": "static",
      "xmlns": "urn:mpeg:DASH:schema:MPD:2011",
      "xmlns:ytdrm": "http://youtube.com/ytdrm",
    },
    "children": Array [
      Object {
        "attributes": Object {
          "duration": "PT0H3M1.63S",
          "start": "PT0S",
        },
        "children": Array [
          Object {
            "attributes": Object {},
            "children": Array [
              Object {
                "attributes": Object {
                  "contentType": "video",
                  "id": "1",
                },
                "children": undefined,
                "name": "ContentComponent",
              },
              Object {
                "attributes": Object {
                  "schemeIdUri": "com.youtube.clearkey",
                },
                "children": Array [
                  Object {
                    "attributes": Object {
                      "key": "1a8a2095e4deb2d29ec816ac7bae2082",
                      "keyid": "60061e017e477e877e57d00d1ed00d1e",
                    },
                    "children": undefined,
                    "name": "ytdrm:License",
                  },
                ],
                "name": "ContentProtection",
              },
              Object {
                "attributes": Object {
                  "bandwidth": "4190760",
                  "codecs": "avc1.640028",
                  "height": "1080",
                  "id": "1",
                  "mimeType": "video/mp4",
                  "width": "1920",
                },
                "children": Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      "car_cenc-20120827-89.mp4",
                    ],
                    "name": "BaseURL",
                  },
                  Object {
                    "attributes": Object {
                      "indexRange": "2755-3230",
                    },
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "range": "0-2754",
                        },
                        "children": undefined,
                        "name": "Initialization",
                      },
                    ],
                    "name": "SegmentBase",
                  },
                ],
                "name": "Representation",
              },
              Object {
                "attributes": Object {
                  "bandwidth": "2073921",
                  "codecs": "avc1.4d401f",
                  "height": "720",
                  "id": "2",
                  "mimeType": "video/mp4",
                  "width": "1280",
                },
                "children": Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      "car_cenc-20120827-88.mp4",
                    ],
                    "name": "BaseURL",
                  },
                  Object {
                    "attributes": Object {
                      "indexRange": "2789-3264",
                    },
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "range": "0-2788",
                        },
                        "children": undefined,
                        "name": "Initialization",
                      },
                    ],
                    "name": "SegmentBase",
                  },
                ],
                "name": "Representation",
              },
              Object {
                "attributes": Object {
                  "bandwidth": "869460",
                  "codecs": "avc1.4d401e",
                  "height": "480",
                  "id": "3",
                  "mimeType": "video/mp4",
                  "width": "854",
                },
                "children": Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      "car_cenc-20120827-87.mp4",
                    ],
                    "name": "BaseURL",
                  },
                  Object {
                    "attributes": Object {
                      "indexRange": "2789-3264",
                    },
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "range": "0-2788",
                        },
                        "children": undefined,
                        "name": "Initialization",
                      },
                    ],
                    "name": "SegmentBase",
                  },
                ],
                "name": "Representation",
              },
              Object {
                "attributes": Object {
                  "bandwidth": "686521",
                  "codecs": "avc1.4d401e",
                  "height": "360",
                  "id": "4",
                  "mimeType": "video/mp4",
                  "width": "640",
                },
                "children": Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      "car_cenc-20120827-86.mp4",
                    ],
                    "name": "BaseURL",
                  },
                  Object {
                    "attributes": Object {
                      "indexRange": "2789-3264",
                    },
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "range": "0-2788",
                        },
                        "children": undefined,
                        "name": "Initialization",
                      },
                    ],
                    "name": "SegmentBase",
                  },
                ],
                "name": "Representation",
              },
              Object {
                "attributes": Object {
                  "bandwidth": "264835",
                  "codecs": "avc1.4d4015",
                  "height": "240",
                  "id": "5",
                  "mimeType": "video/mp4",
                  "width": "426",
                },
                "children": Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      "car_cenc-20120827-85.mp4",
                    ],
                    "name": "BaseURL",
                  },
                  Object {
                    "attributes": Object {
                      "indexRange": "2753-3228",
                    },
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "range": "0-2752",
                        },
                        "children": undefined,
                        "name": "Initialization",
                      },
                    ],
                    "name": "SegmentBase",
                  },
                ],
                "name": "Representation",
              },
            ],
            "name": "AdaptationSet",
          },
          Object {
            "attributes": Object {},
            "children": Array [
              Object {
                "attributes": Object {
                  "contentType": "audio",
                  "id": "2",
                },
                "children": undefined,
                "name": "ContentComponent",
              },
              Object {
                "attributes": Object {
                  "schemeIdUri": "com.youtube.clearkey",
                },
                "children": Array [
                  Object {
                    "attributes": Object {
                      "key": "1a8a2095e4deb2d29ec816ac7bae2082",
                      "keyid": "60061e017e477e877e57d00d1ed00d1e",
                    },
                    "children": undefined,
                    "name": "ytdrm:License",
                  },
                ],
                "name": "ContentProtection",
              },
              Object {
                "attributes": Object {
                  "bandwidth": "127236",
                  "codecs": "mp4a.40.02",
                  "id": "6",
                  "mimeType": "audio/mp4",
                  "numChannels": "2",
                  "sampleRate": "44100",
                },
                "children": Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      "car_cenc-20120827-8c.mp4",
                    ],
                    "name": "BaseURL",
                  },
                  Object {
                    "attributes": Object {
                      "indexRange": "2673-2932",
                    },
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "range": "0-2672",
                        },
                        "children": undefined,
                        "name": "Initialization",
                      },
                    ],
                    "name": "SegmentBase",
                  },
                ],
                "name": "Representation",
              },
              Object {
                "attributes": Object {
                  "bandwidth": "255236",
                  "codecs": "mp4a.40.02",
                  "id": "7",
                  "mimeType": "audio/mp4",
                  "numChannels": "2",
                  "sampleRate": "44100",
                },
                "children": Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      "car_cenc-20120827-8d.mp4",
                    ],
                    "name": "BaseURL",
                  },
                  Object {
                    "attributes": Object {
                      "indexRange": "2673-2932",
                    },
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "range": "0-2672",
                        },
                        "children": undefined,
                        "name": "Initialization",
                      },
                    ],
                    "name": "SegmentBase",
                  },
                ],
                "name": "Representation",
              },
              Object {
                "attributes": Object {
                  "bandwidth": "31749",
                  "codecs": "mp4a.40.03",
                  "id": "8",
                  "mimeType": "audio/mp4",
                  "numChannels": "1",
                  "sampleRate": "22050",
                },
                "children": Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      "car_cenc-20120827-8b.mp4",
                    ],
                    "name": "BaseURL",
                  },
                  Object {
                    "attributes": Object {
                      "indexRange": "2673-2932",
                    },
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "range": "0-2672",
                        },
                        "children": undefined,
                        "name": "Initialization",
                      },
                    ],
                    "name": "SegmentBase",
                  },
                ],
                "name": "Representation",
              },
            ],
            "name": "AdaptationSet",
          },
        ],
        "name": "Period",
      },
    ],
    "name": "MPD",
  },
]
`;
